Source/WTF/wtf/ThreadSafeWeakHashSet.h

```diff
@@ -35,23 +35,74 @@ template<typename T>
 class ThreadSafeWeakHashSet final {
     WTF_MAKE_FAST_ALLOCATED;
 public:
+    ThreadSafeWeakHashSet() = default;
+    ThreadSafeWeakHashSet(ThreadSafeWeakHashSet&& other) { moveFrom(WTFMove(other)); }
+    ThreadSafeWeakHashSet& operator=(ThreadSafeWeakHashSet&& other)
+    {
+        moveFrom(WTFMove(other));
+        return *this;
+    }
+
+    class const_iterator {
+    public:
+        using iterator_category = std::forward_iterator_tag;
+        using value_type = T;
+        using pointer = const value_type*;
+        using reference = const value_type&;
+
+    private:
+        const_iterator(Vector<Ref<T>>&& strongReferences)
+            : m_strongReferences(WTFMove(strongReferences)) { }
+
+    public:
+        T* get() const { return m_position < m_strongReferences.size() ? m_strongReferences[m_position].ptr() : nullptr; }
+        T& operator*() const { return *get(); }
+        T* operator->() const { return get(); }
+
+        const_iterator& operator++()
+        {
+            ASSERT(m_position != m_strongReferences.size());
```

Comment:
I'll add RELEASE_ASSERT(m_position < m_strongReferences.size()); to both.  I think the optimizer will remove one of them, and it will be just as many branches as we have now.